contracts/contracts/MerkleDrop.sol

```diff
@@ -7,38 +7,52 @@ contract MerkleDrop {
 
     bytes32 public root;
     DroppedToken public droppedToken;
+    uint public decayStartTime;
+    uint public decayDurationInSeconds;
 
     mapping (address => bool) withdrawn;
 
     event Withdraw(address recipient, uint value);
 
-    constructor(DroppedToken _droppedToken, bytes32 _root) public {
-        root = _root;
+    constructor(DroppedToken _droppedToken, bytes32 _root, uint _decayStartTime, uint _decayDurationInSeconds) public {
         droppedToken = _droppedToken;
+        root = _root;
+        decayStartTime = _decayStartTime;
+        decayDurationInSeconds = _decayDurationInSeconds;
     }
 
     function withdraw(uint value, bytes32[] memory proof) public {
         withdrawFor(msg.sender, value, proof);
     }
 
     function withdrawFor(address recipient, uint value, bytes32[] memory proof) public {
-        require(checkEntitlement(recipient, value, proof), "The proof could not be verified.");
+        require(verifyEntitled(recipient, value, proof), "The proof could not be verified.");
         require(! withdrawn[recipient], "The recipient has already withdrawn its entitled token.");
-        require(droppedToken.balanceOf(address(this)) > value, "The MerkleDrop does not have tokens to drop yet / anymore.");
+        require(droppedToken.balanceOf(address(this)) >= value, "The MerkleDrop does not have tokens to drop yet / anymore.");
```

Comment:
Ok, I see, this fixes the issue I complained about in the other PR. This would have deserved a dedicated commit. Also the subject you put into the commit messages would have been shorter.

Reply:
At this point I would remove the requirement. It may prevent someone from withdrawing his entitled tokens since he will only receive `valueToSend` tokens.  Please either remove the `require` or replace check the `valueToSend` against the balance.

Reply:
I replaced it with valueToSend. The token contract might not have a clear message as to why the withdraw fails so I would rather provide it in the MerkleDrop.